### ISSUE TITLE
Warn on exit

### ISSUE THREAD

**Issue by helfi92** [taskcluster/taskcluster-web#320]
**Problem**: https://bugzilla.mozilla.org/show_bug.cgi?id=1515233#c0

**Solution**:

One way we could do t hat is to build a utility component that takes a `message` and a `dirty` flag that alerts the user with the `message` whenever the user attempts to leave the page and the `dirty` flag is set to true. Once this is done, we can start integrating it to the views that need it:

* http://localhost:5080/hooks/create
* http://localhost:5080/secrets/create
* http://localhost:5080/tasks/create
* http://localhost:5080/auth/clients/create
* http://localhost:5080/auth/roles/create
* http://localhost:5080/shell
* http://localhost:5080/display

Adding it to one view is good enough for this issue. We could create a new issue later to have this utility be used by the other views that require this functionality.

**Comment by helfi92** [comment]
This task is assigned to @arshadkazmi42. https://bugzilla.mozilla.org/show_bug.cgi?id=1515233.

**Comment by helfi92** [comment]
Don't hesitate to ask me questions if there's anything unclear :)

**Comment by arshadkazmi42** [comment]
@helfi92 I have created a PR for Hooks create page.
This component for warn popup is working, without having dirty flag.
I have mapped `beforeunload` to trigger the warning popup.

Can you review it once.

**Comment by helfi92** [comment]
@arshadkazmi42 Did you want to add ` @withAlertOnClose` to the [other views](https://github.com/taskcluster/taskcluster-web/issues/320#issue-392632322) as well? :)

**Comment by arshadkazmi42** [comment]
Yes. I would like to work on adding this to other views. Will start working on it
